### DISCUSSION TITLE
Feature request: something to test email settings

### DISCUSSION THREAD

**Discussion by tpansino** [cmintey/wishlist#125] [Ideas]
```
As an admin
I want to be able to test that the email settings I've configured in my instance are correct
So that I don't find out later that my emails aren't going through.
```

Most apps I self-host seem to do this with just a `Test email` button of some kind on the same settings page, along with a pop-up dialog that prompts the admin to enter an address to send the test email to, and Submit.

**Comment by cmintey** [comment]
There is a "Test Email" button on the settings page once you hit "Save" and you've filled in the necessary SMTP fields. It doesn't ask for an email and will send it to the email of whichever admin pressed the button
![image](https://github.com/user-attachments/assets/a5373a46-c646-4a58-b8b7-d9ca2f46cb5d)

**Reply by tpansino** [reply]
Ah, thank you!

I guess then my feature request is can that button be displayed permanently once creds are saved? There are times I've found it useful to test whether email is working again, like after rotating the password for the email account.

**Reply by cmintey** [reply]
Whenever you re-save it will pop back up. But I guess it can just always be shown as long as the fields are populated

**Reply by jan-eckhoff** [reply]
Hi there, I just set up Wishlist and I stumbled over how that "Test" button works. As a UX person, I can't help myself from analysing it, so here goes :-D

<img width="851" height="424" alt="image" src="https://github.com/user-attachments/assets/d56f3cf7-a92b-4b2e-a031-53a091fdb2b4" />

Visually, at least on desktop, "Test" is above the "Save" button, so the most likely user flow is to enter the settings, then click "Test".

I did that multiple times, only to get authentication errors all the time. The reason being of course, I should have saved everything before testing it. With the visual order as it stands now, I suppose many users will make that error.

Remedy could be to either change the visual order - make the "Test" button appear below or right of the "Save" button. Or save automatically on testing. Or, probably the most challenging option, test the settings as they are in the input fields right now, irrespective of the save status.

**Reply by cmintey** [reply]
Yeah this is on my radar to fix. There is a issue opened for it already: https://github.com/cmintey/wishlist/issues/587
